src/query_plan/query_planning_traversal.rs

```diff
@@ -87,17 +89,72 @@ struct OpenBranchAndSelections {
     selections: Vec<NormalizedSelection>,
 }
 
-struct BestQueryPlanInfo {
+pub(crate) struct BestQueryPlanInfo {
     /// The fetch dependency graph for this query plan.
-    fetch_dependency_graph: FetchDependencyGraph,
+    pub fetch_dependency_graph: FetchDependencyGraph,
     /// The path tree for the closed branch options chosen for this query plan.
-    path_tree: OpPathTree,
+    pub path_tree: OpPathTree,
     /// The cost of this query plan.
-    cost: QueryPlanCost,
+    pub cost: QueryPlanCost,
 }
 
-impl QueryPlanningTraversal {
-    fn find_best_plan(&mut self) -> Result<Option<&BestQueryPlanInfo>, FederationError> {
+impl BestQueryPlanInfo {
+    // PORT_NOTE: The equivalent of `createEmptyPlan` in the JS codebase.
+    pub fn empty(parameters: &QueryPlanningParameters) -> Self {
+        Self {
+            fetch_dependency_graph: FetchDependencyGraph::new(
+                parameters.supergraph_schema.clone(),
+                parameters.federated_query_graph.clone(),
+                None,
+                0,
+            ),
+            path_tree: OpPathTree::new(parameters.federated_query_graph.clone(), parameters.head),
+            cost: Default::default(),
+        }
+    }
+}
+
+impl<'a> QueryPlanningTraversal<'a> {
+    pub fn new(
+        // TODO(@goto-bus-stop): This probably needs a mutable reference for some of the
```

Comment:
indeed, we have couple of stories left to finish porting the traversal logic - it looks like struct is currently missing some data (including calculation of options etc), if needed can you open up a separate issue to track it?

Reply:
FED-147 ✅